### DISCUSSION TITLE
Minimal* Windscribe example

### DISCUSSION THREAD

**Discussion by randomllama** [qdm12/gluetun#1546] [Q&A]
Hello, I'm moving from a different VPN to Windscribe.  My existing container solution was provider-specific, so I'm setting up Gluetun for the first time.  First time using Wireguard as well.  Thought I'd share my working `compose.yml` files for other fools like me who could not get the Wireguard variant working.

# OpenVPN

```
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
    - NET_ADMIN
    devices:
    - /dev/net/tun
    environment:
      VPN_SERVICE_PROVIDER: windscribe
      SERVER_CITIES: ...
      #> Ref: https://windscribe.com/getconfig/openvpn
      OPENVPN_USER: ...
      OPENVPN_PASSWORD: ...
    networks:
    - internet
networks:
  internet:
```

# Wireguard

```
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
    - NET_ADMIN
    devices:
    - /dev/net/tun
    environment:
      VPN_SERVICE_PROVIDER: windscribe
      SERVER_CITIES: ...
      #> Ref: https://windscribe.com/getconfig/wireguard
      VPN_TYPE: wireguard
      WIREGUARD_ADDRESSES: ...
      WIREGUARD_PRIVATE_KEY: ...
      WIREGUARD_PRESHARED_KEY: ...
      #> Preshared key is required here, despite what the wiki says.
    networks:
    - internet
networks:
  internet:
```

Note that the address Windscribe hands out here is not a private address, but in the [Carrier-grade NAT](https://en.wikipedia.org/wiki/Carrier-grade_NAT) range.  (This confused me greatly until I realized this as I wasn't sure what address to put there.)

_*For certain values of minimal.  I like the verbosity of naming the network connection "internet" instead of using an ambiguous "default".  Is it an public?  Private?  Well, now we know._

Also, I'm clicking the button stating I've done a search without actually doing so, as is internet tradition.  (:

**Comment by porfinsro** [comment]
I was just setting up gluetun with wireguard a few days ago and can confirm that for Windscribe the preshared key is indeed required, not optional - otherwise it won't be able to establish an connection. Took me a few housr to figure that out...

**Comment by hoopahmadness** [comment]
Excellent, I was struggling with the wiki-given compose file but this worked. Thanks!
